File: web/src/lib/labels.ts
// Swedish display labels for backend enum values. Kept in one place so every
// screen shows the same wording.

import type { CircleType, MembershipRole } from "./api";

export const circleTypeLabel: Record<CircleType, string> = {
  Team: "Lag",
  Board: "Styrelse",
  Officials: "Funktionärer",
  General: "Allmän",
};

export const roleLabel: Record<MembershipRole, string> = {
  Player: "Spelare",
  Guardian: "Vårdnadshavare",
  Coach: "Tränare",
  Leader: "Ledare",
  Administrator: "Administratör",
  Member: "Medlem",
};

export function firstName(fullName: string | null): string | null {
  if (!fullName) return null;
  return fullName.trim().split(/\s+/)[0] || null;
}

File: src/Circles.Web/Shared/Labels.cs
using Circles.Domain.Enums;

namespace Circles.Web.Shared;

/// <summary>
/// Swedish display labels for backend enum values. Kept in one place so every
/// screen shows the same wording (ported from the previous frontend).
/// </summary>
public static class Labels
{
    public static string Circle(CircleType type) => type switch
    {
        CircleType.Team => "Lag",
        CircleType.Board => "Styrelse",
        CircleType.Officials => "Funktionärer",
        CircleType.General => "Allmän",
        _ => type.ToString(),
    };

    public static string Role(MembershipRole role) => role switch
    {
        MembershipRole.Player => "Spelare",
        MembershipRole.Guardian => "Vårdnadshavare",
        MembershipRole.Coach => "Tränare",
        MembershipRole.Leader => "Ledare",
        MembershipRole.Administrator => "Administratör",
        MembershipRole.Member => "Medlem",
        _ => role.ToString(),
    };

    /// <summary>Direct vs derived (guardian) access badge text.</summary>
    public static string Access(string accessKind) => accessKind switch
    {
        "Direct" => "Direkt",
        "Derived" => "Härledd",
        _ => accessKind,
    };

    public static string Initials(string? fullName)
    {
        if (string.IsNullOrWhiteSpace(fullName)) return "?";
        var parts = fullName.Trim().Split(' ', StringSplitOptions.RemoveEmptyEntries);
        if (parts.Length == 0) return "?";
        if (parts.Length == 1) return parts[0][..1].ToUpperInvariant();
        return (parts[0][..1] + parts[^1][..1]).ToUpperInvariant();
    }
}
